var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -418,10 +418,10 @@ class Trilinos(CMakePackage, CudaPackage, ROCmPackage):
     depends_on("hwloc+cuda", when="@13: +kokkos+cuda")
     depends_on("hypre@develop", when="@master: +hypre")
     depends_on("netcdf-c+mpi+parallel-netcdf", when="+exodus+mpi@12.12.1:")
+    depends_on("superlu-dist@:4.3", when="@11.14.1:12.6.1+superlu-dist")
     depends_on("superlu-dist@4.4:5.3", when="@12.6.2:12.12.1+superlu-dist")
     depends_on("superlu-dist@5.4:6.2.0", when="@12.12.2:13.0.0+superlu-dist")
-    depends_on("superlu-dist@6.3.0:", when="@13.0.1:99 +superlu-dist")
-    depends_on("superlu-dist@:4.3", when="@11.14.1:12.6.1+superlu-dist")
+    depends_on("superlu-dist@6.3.0:7.99.99", when="@13.0.1:13.99.99 +superlu-dist")
```

Comment:
I believe you could get the same effect from 
`depends_on("superlu-dist@6.3.0:7", when="@13.0.1:13 +superlu-dist")`